keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -18,15 +15,34 @@
 public class BayesianNetwork {
 
     private final List<? extends Vertex> vertices;
+    private final HashMap<String, Vertex> labelSet;
```

Comment:
`Set` is actually a type of collection in Java.  Can this be vertexLabels? Or labelToVertexMap?